tests/api_tests/nodes/test_views.py

```diff
@@ -879,6 +879,514 @@ def test_filtering_node_with_non_bibliographic_contributor(self):
         assert_equal(len(res.json['data']), 1)
         assert_false(res.json['data'][0]['attributes'].get('bibliographic', None))
 
+
+class TestNodeContributorAdd(ApiTestCase):
+    def setUp(self):
+        super(TestNodeContributorAdd, self).setUp()
+        self.user = AuthUserFactory()
+        self.user_two = AuthUserFactory()
+        self.user_three = AuthUserFactory()
+
+        self.private_project = ProjectFactory(creator=self.user, is_public=False)
+        self.private_url = '/{}nodes/{}/contributors/'.format(API_BASE, self.private_project._id)
+
+        self.public_project = ProjectFactory(creator=self.user, is_public=True)
+        self.public_url = '/{}nodes/{}/contributors/'.format(API_BASE, self.public_project._id)
+
+        self.data_user_two = {
+            'id': self.user_two._id,
+            'bibliographic': True
+        }
+
+        self.data_user_three = {
+            'id': self.user_three._id,
+            'bibliographic': True
+        }
+
+    def test_adds_bibliographic_contributor_public_project_admin(self):
+        res = self.app.post_json(self.public_url, self.data_user_two, auth=self.user.auth)
+        assert_equal(res.status_code, 201)
+        assert_equal(res.json['data']['id'], self.user_two._id)
+
+        self.public_project.reload()
+        assert_in(self.user_two, self.public_project.contributors)
+
+    def test_adds_non_bibliographic_contributor_private_project_admin(self):
+        data = {
+            'id': self.user_two._id,
+            'bibliographic': False
+        }
+        res = self.app.post_json(self.private_url, data, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 201)
+        assert_equal(res.json['data']['id'], self.user_two._id)
+
+        self.private_project.reload()
+        assert_in(self.user_two, self.private_project.contributors)
+        assert_false(self.private_project.get_visible(self.user_two))
+
+    def test_adds_contributor_public_project_non_admin(self):
+        self.public_project.add_contributor(self.user_two, permissions=['read', 'write'], auth=Auth(self.user))
+        self.public_project.reload()
```

Comment:
The reload should happen *after* the post request, not before.

But given that the model is not saved after adding the contributor (no kwarg `save=True`), I don't think this test is failing for quite the expected reason. Worth a closer look.